shared/actions/tracker.js

```diff
@@ -425,16 +419,43 @@ function updatePGPKey (username: string, pgpFingerprint: Buffer, kid: string): A
 const sessionIDToUsername: { [key: number]: string } = {}
 // TODO: if we get multiple tracker calls we should cancel one of the sessionIDs, now they'll clash
 function serverCallMap (dispatch: Dispatch, getState: Function, skipPopups: boolean = false, onFinish: ?() => void): incomingCallMapType {
+  // if true we already have a pending call so lets skip a ton of work
+  let username
+  let clearPendingTimeout
+  let alreadyPending = false
+
+  const requestIdle = f => {
```

Comment:
just a wrapper so we check if its pending and only do 'our' work if its not a duplicate (we always respond to the server)